content-audit/kafka-publisher/src/main/java/org/commonjava/propulsor/KafkaPublisher.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2011-2018 Red Hat, Inc. (https://github.com/Commonjava/propulsor)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.commonjava.propulsor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.commonjava.propulsor.content.audit.FileEventPublisher;
+import org.commonjava.propulsor.content.audit.FileEventPublisherException;
+import org.commonjava.propulsor.content.audit.model.FileEvent;
+import org.commonjava.propulsor.content.audit.model.FileGroupingEvent;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Properties;
+
+public class KafkaPublisher
+                implements FileEventPublisher
+{
+
+    private static final String PROP_PATH = "publisher.properties";
+
```

Comment:
Looks i can define a config class that with annotation like @SectionName( "publisher.kafka" ) and move the config into the unified file.